pkgs/games/eternity-engine/default.nix

```diff
@@ -1,20 +1,21 @@
-{ lib, stdenv, cmake, libGL, SDL, SDL_mixer, SDL_net, fetchFromGitHub, makeWrapper }:
+{ lib, stdenv, cmake, libGL, SDL2, SDL2_mixer, SDL2_net, fetchFromGitHub, makeWrapper }:
 
 stdenv.mkDerivation rec {
   pname = "eternity-engine";
-  version = "3.42.02";
+  version = "4.02.00";
   src = fetchFromGitHub {
     owner = "team-eternity";
     repo = "eternity";
     rev = version;
-    sha256 = "00kpq4k23hjmzjaymw3sdda7mqk8fjq6dzf7fmdal9fm7lfmj41k";
+    sha256 = "0dlz7axbiw003bgwk2hl43w8r2bwnxhi042i1xwdiwaja0cpnf5y";
+    fetchSubmodules = true;
```

Comment:
Please devendor https://github.com/Wohlstand/libADLMIDI/ and add it as an extra package.

Reply:
The version of ADLMIDI that eternity includes as a submodule doesn't correspond to any git tag in the ADLMIDI repo. It appears to be ahead of the latest git tag. So a separate ADLMIDI package would have to be based on that specific git hash that eternity wants. Also the eternity build instructions don't explain how to build against a pre-built ADLMIDI. So is it worth splitting up like this?